co-log/src/Colog/Message.hs

```diff
@@ -0,0 +1,45 @@
+{-# LANGUAGE FlexibleContexts #-}
+
+module Colog.Message
+       ( LogMessage (..)
+       , log
+       , logDebug
+       , logInfo
+       , logNotice
+       , logWarning
+       , logError
+
+       , fmtLogMessage
+       ) where
+
+import Colog.Core.Severity (Severity (..))
+import Colog.Monad (WithLog, logMsg)
+
+data LogMessage = LogMessage Text Severity
```

Comment:
What do you think about renaming this data type from `LogMessage` to just `Message`? I think this looks nice in constraints:

```haskell
WithLog env Message m => ...
```

And we can have nice record data type:

```haskell
data Message = Message
    { messageSeverity :: {-# UNPACK #-} !Severity
    , messageText     :: {-# UNPACK #-} !Text
    } deriving ... something useful ... ?
```

(those `{-# UNPACK #-}` and `!` annotations are for performance, because we want performant logging library)